src/base/types.ts

```diff
@@ -0,0 +1,194 @@
+import { RedisClientOptions, createClient } from "redis";
+import Queue from "./Queue";
+
+/* ========== Interfaces ========== */
+
+/**
+ * Log method which accepts a message and optionally the metadata
+ */
+type ILogMethod = (message: string, meta?: any) => any;
+
+/**
+ * Accepted logger interface
+ */
+export interface ILogger {
+  debug: ILogMethod;
+  error: ILogMethod;
+  warn: ILogMethod;
+  info: ILogMethod;
+  verbose: ILogMethod;
+}
+
+/**
+ * Has connect, disconnect, start and stop method
+ */
+export interface IStartable {
```

Comment:
Nit: StartStoppable? (I really expected only a start() method here.)

Reply:
Yeah, first I created IConnectable and IStartable, they I realized we also need IDisconnectable and IStoppable too, so just went with this one haha